crates/apollo-parser/src/lexer/mod.rs

```diff
@@ -254,12 +255,11 @@ impl<'a> Cursor<'a> {
                 },
                 State::StringLiteralEscapedUnicode(remaining) => match c {
                     '"' => {
-                        self.add_err(Error::new(
+                        return Err(Error::with_loc(
                             "incomplete unicode escape sequence",
-                            c.to_string(),
+                            self.current_str().to_string(),
+                            token.index,
                         ));
-                        token.data = self.current_str();
-                        return self.done(token);
```

Comment:
This is a slight behaviour change, because we would previously still add the string token (knowing it's invalid, but that is intentional in apollo-parser). Was there some issue with using `.add_err()` here?

Reply:
Ah yes you're right. Sorry, I didn't mean to change the behaviour, I just felt like this was more consistent with the rest of the code. I will revert this part.

I will also remove `Error::new`.